File: backend/domain/entity/timetable.go
package entity

import "github.com/tsuna-can/express-bus-time-table-app/backend/domain/vo"

type Timetable struct {
	ParentRouteId    string
	ParentRouteName  vo.ParentRouteName
	BusStopId        string
	BusStopName      vo.BusStopName
	TimetableEntries []TimetableEntry
}

func NewTimetable(
	parentRouteId string,
	parentRouteName vo.ParentRouteName,
	busStopId string,
	busStopName vo.BusStopName,
	timetableEntries []TimetableEntry) *Timetable {
	return &Timetable{
		ParentRouteId:    parentRouteId,
		ParentRouteName:  parentRouteName,
		BusStopId:        busStopId,
		BusStopName:      busStopName,
		TimetableEntries: timetableEntries,
	}
}

File: frontend/app/src/main/java/com/tsunacan/expressbustimetableapp/models/Timetable.kt
package com.tsunacan.expressbustimetableapp.models

/**
 * Time table for a bus stop
 */
data class Timetable(
    val parentRouteId: String = "",
    val parentRouteName: String = "",
    val stopId: String = "",
    val stopName: String = "",
    val timetableEntryList: List<TimetableEntry> = emptyList()
)
